u2f-gae-demo/src/com/google/u2f/gaedemo/storage/TokenStorageData.java

```diff
@@ -54,22 +52,24 @@ public void updateCounter(int newCounterValue) {
   }
 
   public SecurityKeyData getSecurityKeyData() {
-    X509Certificate x509cert = parseCertificate(attestationCert);
-    return new SecurityKeyData(enrollmentTime, transports, keyHandle, publicKey, x509cert, counter);
+    X509Certificate[] x509certChain = parseCertificateChain(attestationCert);
+    return new SecurityKeyData(
+        enrollmentTime, transports, keyHandle, publicKey, x509certChain, counter);
   }
 
   public JsonObject toJson() {
-    X509Certificate x509cert = getSecurityKeyData().getAttestationCertificate();
+    X509Certificate[] x509certChain = getSecurityKeyData().getAttestationCertificateChain();
     JsonObject json = new JsonObject();
     json.addProperty("enrollment_time", enrollmentTime);
     json.add("transports", getJsonTransports());
     json.addProperty("key_handle", Hex.encodeHexString(keyHandle));
     json.addProperty("public_key", Hex.encodeHexString(publicKey));
-    json.addProperty("issuer", x509cert.getIssuerX500Principal().getName());
+    json.addProperty("issuer", x509certChain[0].getIssuerX500Principal().getName());
 
     try {
+      // TODO(aczeskis): use the actual root ca cert when it's available
       AndroidKeyStoreAttestation androidKeyStoreAttestation =
-          AndroidKeyStoreAttestation.Parse(x509cert);
+          AndroidKeyStoreAttestation.Parse(x509certChain, null);
```

Comment:
My feeble brain can't parse null without some annotation. Could you please comment what it means here?